### PR TITLE
added json output for branches script

### PR DESCRIPTION
**Pull request by bflanagin**
First step in adding automated test building based on PRs in openQA. 

Usage:  `./scripts/branches --output > branches.json`

### CODE REVIEW

**Comment by jacobgkau** [comment]
This is working, but the syntax seems a little off. The usage shown in the PR description does not work.

`./scripts/branches --output > branches.json` returns this output:
```
usage: branches [-h] [--output OUTPUT] [repos ...]
branches: error: argument --output: expected one argument
```

`./scripts/branches --output branches.json` returns the entire json to stdout and saves nothing to the json file. So the actual command needed is `./scripts/branches --output branches.json > branches.json`, which is unintuitive.

Can we do the following?

- Have a `--json` option that changes the format to json.
- Either have a separate `--output` option that does not require redirecting stdout, or remove that option and just rely on redirection instead.

**Comment by bflanagin** [comment]
I like the --json option. I was considering adding other output options, but for now I think the json output is good enough.

File: scripts/branches [hunk 1]
```diff
@@ -4,20 +4,32 @@ import argparse
...
+	else:
+		out[repo["name"]] = []
+	for branch in github(repo["branches_url"].replace("{/branch}", "")):
+		if branch["name"] == repo["default_branch"]:
+			if args.output == None:
+				print(" * " + branch["name"])
+			else:
+				out[repo["name"]].append(branch["name"])
+		else:
+			if args.output == None:
```
Comment (13r0ck) [inline scripts/branches:28]: This else clause is almost identical to the previous. This should be possible without nesting nearly identical if statements.

**Review by 13r0ck — CHANGES_REQUESTED** [review]

**Comment by 13r0ck** [comment]
Not necessary, but it might be nice if this also checked environment variables, in the case this is ever used on a cloud server, nearly all cloud providers make secret values available to the container via environment variables.

File: scripts/branches (hunk 1 above)
Comment (13r0ck) [reply scripts/branches:28]: Probably something like 
```
spacer = " * " if branch["name"] == repo["default_branch"] else "   "
if args.output == None:
    print(spacer + branch["name"])
else:
    out[repo["name"]].append(branch["name"])
```

**Comment by 13r0ck** [comment]
Just out of curiosity, is it possible to skip the github key by just adding a sleep time to the script between making requests. Obviously would take longer, and I'm not sure what github's rate limiting is like, so is that even possible?

**Comment by bflanagin** [comment]
@13r0ck Unfortunately github is smarter that that. After the limit is reached you're out of luck for at least 30 minutes. Could be longer but I didn't feel like waiting and just added the key.,

File: scripts/branches (hunk 1 above)
Comment (bflanagin) [reply scripts/branches]: Already changed, :+1:

**Comment by bflanagin** [comment]
Command should output a json formatted list if you use `branches --json`

**Comment by bflanagin** [comment]
@jacobgkau I missed the part about adding the output option instead of using stdout. I don't need the script to create a file for my use, but thought it was the easiest way to test the output while testing. I'll be more verbose in future PRs so that the task and the goal is clear.

**Review by jacobgkau — DISMISSED** [review]
Output without `--json` is identical to how it was before, output with `--json` prints in json.

I did notice while looking at the output that it doesn't look like the json version indicates the default branch like the normal output does? But if it's not needed for the current use case, I don't think that's a problem. (It would obviously need to be presented separately from the list in json, rather than using an asterisk like the normal output uses.)

**Review by 13r0ck — CHANGES_REQUESTED** [review]

File: scripts/branches (hunk 1 above)
Comment (13r0ck) [reply scripts/branches]: Setting the `spacer` or whatever you want to call it is still shorter, but I'm not going to nit pick too much. What you had before wasn't a styleistic choice, what was there before was just bad not DRY code.

**Comment by 13r0ck** [comment]
Does this code break if you have a github key in more than one of the locations?

File: scripts/branches (hunk 1 above)
Comment (bflanagin) [reply scripts/branches]: That's fair

**Comment by bflanagin** [comment]
> Does this code break if you have a github key in more than one of the locations?
No, it breaks the for loop when the first .github_token file is found.

**Comment by bflanagin** [comment]
@13r0ck You're right about the need for lib.py to accept the github_token via environment variable or other means. As myself the changes work fine, but when added to a service it has issues. This PR is more about the output so I'll add the better token search to the next PR.

**Comment by 13r0ck** [comment]
@bflanagin sounds good to me.
Your print statements are intended one tab too far. Fix that then ill approve

**Review by 13r0ck — DISMISSED** [review]

**Comment by 13r0ck** [comment]
Request review:
Your last commit message uses the wrong `they're`, it should be `their` :zany_face:

**Comment by jacobgkau** [comment]
Amended last commit to correct the grammar mistake.

**Review by jacobgkau — APPROVED** [review]
New flag is still working.

**Comment by bflanagin** [comment]
Heh, thanks for changing that @jacobgkau

**Review by 13r0ck — APPROVED** [review]

**Pull request merged**